#!/usr/bin/env python
PACKAGE = "map_transform"

import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("infl", int_t, 0, "Robot size in cells", 5, 1,50)
gen.add("defl", int_t, 0, "Sensing radius in cells", 3, 1, 500)
gen.add("debug", bool_t, 0, "Debugging with xy position parameters", True)
gen.add("show", bool_t, 0, "Show images", True)
gen.add("ground_truth", bool_t, 0, "Ground Truth", False)
gen.add("pub_once", bool_t, 0, "Publish only once", True)
gen.add("frga", bool_t, 0, "Find reachable points that generate actuation points", False)
gen.add("city", bool_t, 0, "Different Nav and Perc maps", False)
gen.add("fast_opt", bool_t, 0, "Fast Visibility, Sub Optimal PA*", False)


gen.add("x", double_t, 0, "X Position", 50, 0, 100)
gen.add("y", double_t, 0, "Y Position", 50, 0, 100)

exit(gen.generate(PACKAGE, "visibility", "Parameters"))
